payments-core/src/main/java/com/stripe/android/cards/CardAccountRangeService.kt

```diff
@@ -0,0 +1,105 @@
+package com.stripe.android.cards
+
+import androidx.annotation.RestrictTo
+import androidx.annotation.VisibleForTesting
+import com.stripe.android.model.AccountRange
+import com.stripe.android.model.CardBrand
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class CardAccountRangeService constructor(
+    private val cardAccountRangeRepository: CardAccountRangeRepository,
+    private val workContext: CoroutineContext,
+    val staticCardAccountRanges: StaticCardAccountRanges,
+    private val accountRangeResultListener: AccountRangeResultListener
+) {
+
+    val isLoading: Flow<Boolean> = cardAccountRangeRepository.loading.map {
+        it
+    }
+    var accountRange: AccountRange? = null
+        private set
+
+    @VisibleForTesting
+    var accountRangeRepositoryJob: Job? = null
+
+    fun onCardNumberChanged(cardNumber: CardNumber.Unvalidated) {
+        val staticAccountRange = staticCardAccountRanges.filter(cardNumber)
+            .let { accountRanges ->
+                if (accountRanges.size == 1) {
+                    accountRanges.first()
+                } else {
+                    null
+                }
+            }
```

Comment:
```suggestion
        val staticAccountRange = staticCardAccountRanges.filter(cardNumber).firstOrNull()
```

Reply:
We need the check for whether the accountRange.size is equal to 1